lib/OpenQA/Worker/Job.pm

```diff
@@ -699,7 +699,7 @@ sub _upload_results_step_0_prepare {
             or $file_info[9] != $self->{_test_order_mtime}
             or $file_info[7] != $self->{_test_order_fsize})
         {
-            log_info('Test schedule has changed, reloading test_order.json');
+            log_info('Test schedule has changed, reloading test_order.json') if $self->{_test_order_mtime};
             $test_order                = $self->_read_json_file('test_order.json');
             $status{test_order}        = $test_order;
```

Comment:
Now it looks like we still update the test order unconditionally, just not log about it. This does not make sense to me. The log call should not have a different logical execution path

Reply:
But when loading the test order *initially* this block must be executed. I don't understand your point.

Reply:
Yes, of course. My point is that the logic differs for the actual code and the log message. IMHO a log message should not be in a different logic branch. E.g.

```perl
if ($self->{_test_order_mtime}) {
    log_info(…);
    do_something;
}
```

Reply:
Then we will never initialize the schedule.
Would you be ok with something like this?
```perl
if ($self->{_test_order_mtime}) {
    log_info('Test schedule has changed, reloading test_order.json');
}
else {
    log_info('Initializing test schedule, loading test_order.json');
}
```

Reply:
but then we would be back with many log messages. You can change one to `log_debug` though.

Now I wonder: Why do we need to repeatedly re-read the json file if nothing changes?

Reply:
This is only executed if the size/mtime of the file changed or if `$current_test_name` is false.

So from my understanding it should only be executed at the beginning and if the file changed.
If `$current_test_name` can be false in between, then yes, that should be changed.

Reply:
See #2445